airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -344,3 +337,34 @@ def _remove_concurrent_streams_from_catalog(
                 if stream.stream.name not in concurrent_stream_names
             ]
         )
+
+    def _retriever_factory(
+        self, stream_config: ComponentDefinition, source_config: Config, stream_state: StreamState
+    ) -> Callable[[], Retriever]:
+        def _factory_method() -> Retriever:
+            declarative_stream: DeclarativeStream = self._constructor.create_component(
```

Comment:
I feel pretty mixed about the fact that we have to create a runtime class of `DeclarativeStream` from the model instead of just instantiating a `SimpleRetriever` runtime instance. Given that I think in a previous convo, it seems reasonable that we want to deprecate `DeclarativeStream` in favor of `DefaultStream`. Similar to the existing code where we don't rely on the low-code runtime component of `DatetimeBasedCursor`, it still feels relevant here too

I presume the reason why we're opting for instantiating the the runtime stream object instead of the retriever directly because we need fields from the DeclarativeStream passed down into the retriever? If so let's just add a comment mentioning that.

Reply:
> I presume the reason why we're opting for instantiating the the runtime stream object instead of the retriever directly because we need fields from the DeclarativeStream passed down into the retriever?

`create_declarative_stream` has some logic to generate the retriever. When the DeclarativeStream will be deleted, this logic will not be deleted but will be used to instantiate a `DefaultStream` with a `StreamSlicerPartitionGenerator`. I haven't put too much time in doing that because today the `client_side_incremental_sync` and `cursor_used_for_stop_condition` depends on the previous interfaces which is blocked by https://github.com/airbytehq/airbyte-internal-issues/issues/10568